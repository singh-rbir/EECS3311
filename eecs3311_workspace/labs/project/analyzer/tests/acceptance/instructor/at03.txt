add_class("A")

-- Error: adding an existing class
add_class("A")

-- Error: the context class does not exist

-- Add an attribute `int att` in class `B`.
add_attribute("B", "att", "INTEGER")

-- Add a command `c()` in class `B`.
add_command("B", "c", <<>>)

-- Add a query `boolean q()` in class `B`.
add_query("B", "q", <<>>, "BOOLEAN")

-- Add the assignment instruction `x = ?` to routine `c` in class `B`.
-- Here routine `c` and variable `x` do not exist, but with a lower priority.
add_assignment_instruction("B", "c", "x")

-- Add a feature with name `f` into class `A`.
add_attribute("A", "f", "INTEGER")

-- Add another feature with clashing name `f` into class `A`.
add_attribute("A", "f", "BOOLEAN") -- add `boolean f`
add_command("A", "f", <<>>) -- add `f()`
add_query("A", "f", <<>>, "A") -- add `A f()`