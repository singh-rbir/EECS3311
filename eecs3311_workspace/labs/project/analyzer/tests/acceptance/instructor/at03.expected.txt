  Status: OK.
  Number of classes being specified: 0
->add_class("A")
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_class("A")
  Status: Error (A is already an existing class name).
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_attribute("B","att","INTEGER")
  Status: Error (B is not an existing class name).
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_command("B","c",<<>>)
  Status: Error (B is not an existing class name).
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_query("B","q",<<>>,"BOOLEAN")
  Status: Error (B is not an existing class name).
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_assignment_instruction("B","c","x")
  Status: Error (B is not an existing class name).
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_attribute("A","f","INTEGER")
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + f: INTEGER
      Number of queries: 0
      Number of commands: 0
->add_attribute("A","f","BOOLEAN")
  Status: Error (f is already an existing feature name in class A).
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + f: INTEGER
      Number of queries: 0
      Number of commands: 0
->add_command("A","f",<<>>)
  Status: Error (f is already an existing feature name in class A).
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + f: INTEGER
      Number of queries: 0
      Number of commands: 0
->add_query("A","f",<<>>,"A")
  Status: Error (f is already an existing feature name in class A).
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + f: INTEGER
      Number of queries: 0
      Number of commands: 0
